### DISCUSSION TITLE
using Micropython on my circuit design

### DISCUSSION THREAD

**Discussion by Hila911** [micropython/micropython#9262] [Using MicroPython]
Hi everyone,
I’m new in micropython and I’m trying to figure out how to use it on my own board. I’ve managed to use the .dfu file one STM32F4DISCO but my board has STM32L4R9 mcu, and doesn’t have USB connection, just a JLink. 
So here are my questions
1. What’s the difference between the a dfu and a bin file created with MakeFile, and how to use the last one?
2. Is there an option to use something similar to putty/picocom without the usb connection?
3. Since I’m having a different pinout in my board, what’s the best way to change those settings (the LED pin for example)?
4. My MCU is different then the one on the EVB, can I still use the same dfu file?

I would extremely appreciate any kind of help, thank you :)

**Comment by jimmo** [comment]
> Hi everyone, I’m new in micropython and I’m trying to figure out how to use it on my own board. I’ve managed to use the .dfu file one STM32F4DISCO but my board has STM32L4R9 mcu, and doesn’t have USB connection, just a JLink. So here are my questions
> 
> 1. What’s the difference between the a dfu and a bin file created with MakeFile, and how to use the last one?

It depends what you're using to flash the microcontroller. .dfu files can be used tools like dfu-util in conjunction with the ST bootloader.

.bin files are just the raw bytes. You also need to tell the programmer what address to put it at. Works with st-link too.

.hex files are another common option, used by the STCubeProgrammer tool (and many more). Newer st-link works with this too via their mass storage mode.

> 2. Is there an option to use something similar to putty/picocom without the usb connection?

You just need a USB UART adaptor (e.g. FTDI breakout) to connect it to your PC.

> 3. Since I’m having a different pinout in my board, what’s the best way to change those settings (the LED pin for example)?

You need to write a board definition. e.g. here's the one for the f4 disco you're using. https://github.com/micropython/micropython/tree/master/ports/stm32/boards/STM32F4DISC

> 4. My MCU is different then the one on the EVB, can I still use the same dfu file?

Nope. Not only are they a different series (F4 vs L4) but generally different parts in the same family will need different firmware. Even different models with the same number (e.g. F405*) can have different RAM/Flash sizes.

FWIW, I'm not sure if MicroPython currently supports the L4+ series.

> 
> I would extremely appreciate any kind of help, thank you :)
